qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/UserAccountSteps.java

```diff
@@ -139,11 +139,44 @@ public void enterDoctorDetails() {
 	@And("a user clicks on save account button")
 	public void clickOnSaveUserAccountButton() {
 		userAccountPage.saveUserAccount();
-		userAccountPage.waitForPage();
 	}
 
 	@Then("the system adds user account into the users table")
 	public void systemAddsUserAccount() {
 		assertTrue(textExists("Account Saved Successfully"));
 	}
+	
+	@When("a user enters person details in the user account form") 
+    	public void enterPersonalDetails() {
+        	userAccountPage.enterPersonalDetails("Clerk", "Data");
+        	userAccountPage.selectGender();
+    	}
+
+    	@And("a user enters user account details in the user account form")
+     	public void enterUserAccountDetails() {
+         	userAccountPage.clickOnAddUserAccount();
+         	userAccountPage.setUsername(CLERK);
+         	userAccountPage.setUserPrivilegeLevel("Full");
+     	}
+
+    	@And("a user enters password that meets the password criteria")
+     	public void userEnterPasswordThatMeetsThePasswordCriteria() {
+         	userAccountPage.setUserPassword("Dataclerk123", "Dataclerk123");
+     	}
+
+     	@And("a user enters password that doesn't meet the password criteria") 
+     	public void userEntersPasswordThatDoesntMeetThePasswordCriteria() {
+         	userAccountPage.setUserPassword("DATACLERK!23", "DATACLERK!23");
+     	}
+     	
+     	@Then("the system confirms the password meets the password criteria")
+     	public void systemConfirmsPasswordMeetsPasswordCriteria() {
+          	assertTrue(textExists("Account Saved Successfully"));
```

Comment:
Which text is displayed on the UI when entered password does not meet the creterial?

Reply:
Validation errors found Failed to save account details")

Reply:
yes that is the error i get when i try to save a user whose password doesn't meet the criteria.

Reply:
I thought the error shows up right at the time when the user enter the password in the field not at the time of saving?

Reply:
it throws `At least 8 character(s) are required` as you type in the field until the password is more than 8 characters!...then `passwords don't match` if the passwords in the two fields dont match.

And lastly it throws `Validation errors found Failed to save account details` if the password generally doesnt meet the criteria e.g password must have atleast a number and a capital with a balance with other characters as shown here --> [where the configurations are made](https://qa-refapp.openmrs.org/openmrs/admin/maintenance/globalProps.form)

Reply:
We can assert that _At least 8 character(s) are required_ does not appear when a user type in a qualified password.

Reply:
> > text_At least 8 character(s) are required_
> 
> Is this the validation error message that appears when password doesnt meet the creteria ?

this only happens during typing of the passwords and not when trying to submit the form so its a no